packages/message-parser/src/grammar.pegjs

```diff
@@ -667,27 +543,84 @@ unicodeEmojiDingbats = $([\u2700-\u27BF] [\uFE00-\uFE0F]?)
 
 unicodeEmojiFlags = $([\uD83C] [\uDD00-\uDDFF] [\uD83C] [\uDD00-\uDDFF])
 
-/* Colors */
+/**
+ *
+ * Inline Code
+ * e.g: `console.log('hello world')`
+ *
+ */
+InlineCode = "`" text:$InlineCode__+ "`" { return inlineCode(plain(text)); }
+
+InlineCode__ = $(!"`" !"\n" .)
 
-color
-  = & { return options.colors; } "color:#" rgba:colorRGBATuple !anyText {
+/**
+ *
+ * Colors
+ * e.g: color:#ff0000 , color:#ff0
+ *
+ */
+color = & { return options.colors; } "color:#" rgba:colorRGBATuple !anyText {
       return color(...rgba);
     }
 
-colorRGBATuple
-  = r:hexByte g:hexByte b:hexByte a:hexByte { return [r, g, b, a]; }
-  / r:hexByte g:hexByte b:hexByte { return [r, g, b]; }
-  / r:hexNible g:hexNible b:hexNible a:hexNible { return [r, g, b, a]; }
-  / r:hexNible g:hexNible b:hexNible { return [r, g, b]; }
+colorRGBATuple = hexByte|3..4| / hexNible|3..4|
+
+/**
+ *
+ * Macros
+ *
+ */
+_ = (EndOfLine / Space)*
+
+Whitespace = w:$Space+ { return plain(w); }
+
+Escaped = "\\" t:[*_~`#.] { return plain(t); }
+
+Any = !EndOfLine t:. p:$AutolinkedPhone? u:$URL? { return plain(t + p + u); }
+
+Line = t:line { return plain(t); }
+line = head:$Space* text:$anyText+ tail:$Space* { return head + text + tail; }
+
+Text = text:anyText { return plain(text); }
+
+EndOfLine = "\r\n" / "\n" / "\r"
+
+Space = " " / "\t"
+
+utf8_names_validation = $([-_.] / alphaChar / decimalNumberChar)+
+
+anyText = [\x20-\x27\x2B-\x40\x41-\x5A\x61-\x7A] / nonascii
+
+nonascii = [\x80-\uFFFF]
+
+unicode = "\\" digits:$(hexdigit |1..6|) ("\r\n" / [ \t\r\n\f])? { return String.fromCharCode(parseInt(digits, 16)); }
+
+alpha = [a-zA-Z]
+
+digit = [0-9]
+
+alpha_digit = [a-zA-Z0-9]
+
+digits = $digit+
+
+safe = [$@&+\__#?-]
+
+extra = [.,!%~*\"':;()=~]
+
+hexdigit = [0-9A-Fa-f]
+
+hexNible = a:hexdigit { return parseInt(a + a, 16); }
+
+hexByte = a:hexdigit b:hexdigit { return parseInt(a + b, 16); }
+
+alphaNumericOrMarkChar = alphaOrMarkChar / decimalNumberChar
 
-/* Tasks */
+alphaOrMarkChar = alphaChar / emojiChar / markChar
 
-tasks = items:task+ { return tasks(items); }
+decimalNumberChar = [0-9\u0660-\u0669\u06F0-\u06F9\u07C0-\u07C9\u0966-\u096F\u09E6-\u09EF\u0A66-\u0A6F\u0AE6-\u0AEF\u0B66-\u0B6F\u0BE6-\u0BEF\u0C66-\u0C6F\u0CE6-\u0CEF\u0D66-\u0D6F\u0DE6-\u0DEF\u0E50-\u0E59\u0ED0-\u0ED9\u0F20-\u0F29\u1040-\u1049\u1090-\u1099\u17E0-\u17E9\u1810-\u1819\u1946-\u194F\u19D0-\u19D9\u1A80-\u1A89\u1A90-\u1A99\u1B50-\u1B59\u1BB0-\u1BB9\u1C40-\u1C49\u1C50-\u1C59\uA620-\uA629\uA8D0-\uA8D9\uA900-\uA909\uA9D0-\uA9D9\uA9F0-\uA9F9\uAA50-\uAA59\uABF0-\uABF9\uFF10-\uFF19]
```

Comment:
I suspect these merged character classes are the main reason why performance got so much better

Reply:
The `decimalNumberChar` or the `tasks`? 
The `decimalNumberChar` was already like that... 

Maybe the perf increase was caused by some use of `count`, `min` and `max` rules, rewrites of some logic, simplifications of `AND` characters operations using regex (e.g:  `safe`, `extra` and etc...)

I'm still trying to find ways to improve and I'm investigating a better way to build emojis/emoticons as you suggested!

BTW, @polkovnikov-ph  thanks a lot for your contribution and your interest

Reply:
No problem at all. It didn't take much time, and I'm a huge fan of user interfaces that are fast and responsive :)